### PR TITLE
Update supabase and postgresql tutorials for st.connection

### PR DESCRIPTION
**Pull request by snehankekre**
## 📚 Context

Updated the Supabase connection tutorial to use `st.experimental_connection` and the [community-built](https://streamlit.io/components?category=connections) connection: [Streamlit Supabase Connector](https://github.com/SiddhantSadangi/st_supabase_connection/tree/main). Also updated the postgres tutorial to use st.connection with the postgresql dialect.

## 🧠 Description of Changes

- Updated the supabase and postgresql tutorials to use the st.connection API

**Revised:**

Here's the main update, there are some other updates too

image

image

**Current:**

image

image

## 💥 Impact

Size:

- [ ] Small 
- [x] Not small 

## 🌐 References

- [x] [Notion](https://www.notion.so/snowflake-corp/Overdue-Use-st-experimental_connection-in-connection-tutorials-f39a02fa0889499dadbce8fb381d1e09)

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

### CODE REVIEW

File: content/kb/tutorials/databases/supabase.md [hunk 1]
```diff
@@ -108,23 +123,14 @@ Copy the code below to your Streamlit app and run it.
...
+supabase = st.experimental_connection("supabase",type=SupabaseConnection)
 
 # Perform query.
-# Uses st.cache_data to only rerun when the query changes or after 10 min.
-@st.cache_data(ttl=600)
 def run_query():
-    return supabase.table("mytable").select("*").execute()
+    return supabase.query("*", table="mytable", ttl=600).execute()
 
 rows = run_query()
```
Comment (sfc-gh-jcarroll) [inline content/kb/tutorials/databases/supabase.md:135]: request that we update this section to roughly the following, unless there's a good reason to keep it this way:

```python
conn = st.experimental_connection("supabase", type=SupabaseConnection)

# Perform query.
rows = conn.query("*", table="mytable", ttl="10m").execute()
```

**Review by sfc-gh-jcarroll — APPROVED** [review]
Requested one change otherwise LGTM

File: content/kb/tutorials/databases/postgresql.md [hunk 2]
```diff
@@ -36,17 +36,18 @@ Your local Streamlit app will read secrets from a file `.streamlit/secrets.toml`
...
+port = "5432"
+database = "xxx"
+username = "xxx"
 password = "xxx"
 ```
 
 <Important>
 
-When copying your app secrets to Streamlit Community Cloud, be sure to replace the values of **host**, **port**, **dbname**, **user**, and **password** with those of your _remote_ PostgreSQL database!
+When copying your app secrets to Streamlit Community Cloud, be sure to replace the values of **host**, **port**, **database**, **user**, and **password** with those of your _remote_ PostgreSQL database!
```
Comment (sfc-gh-dmatthews) [inline content/kb/tutorials/databases/postgresql.md:50]:
```suggestion
When copying your app secrets to Streamlit Community Cloud, be sure to replace the values of **host**, **port**, **database**, **username**, and **password** with those of your _remote_ PostgreSQL database!
```

File: content/kb/tutorials/databases/postgresql.md [hunk 3]
```diff
@@ -58,13 +59,14 @@ As the `secrets.toml` file above is not committed to GitHub, you need to pass it
 
 ![Secrets manager screenshot](/images/databases/edit-secrets.png)
 
-## Add psycopg2 to your requirements file
+## ## Add dependencies to your requirements file
```
Comment (sfc-gh-dmatthews) [inline content/kb/tutorials/databases/postgresql.md:62]:
```suggestion
## Add dependencies to your requirements file
```

**Review by sfc-gh-dmatthews — APPROVED** [review]

File: content/kb/tutorials/databases/postgresql.md (hunk 2 above)
Comment (snehankekre) [reply content/kb/tutorials/databases/postgresql.md:50]: Good catch!

File: content/kb/tutorials/databases/supabase.md (hunk 1 above)
Comment (snehankekre) [reply content/kb/tutorials/databases/supabase.md:135]: Much, better! Thanks 😄

**Pull request merged**